### PR TITLE
HPCC-25746 React version of Landing Zone

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

Create react components for landing zone in ECL watch. Squashed commit messages below

- adding routes / links to react and "legacy" landing zone pages
- fix issue preventing LZBrowseWidget from rendering
- add TargetDropzoneField component
- preview button click, change url to route landingzone preview
- download and delete button actions
- add target-server field type
- change target-dropzone field type after an earlier rebase
- fix some column defintions
- correct Fields and Filter import paths
- wip, test updating prop dropdown dependant upon another
- wip, try overriding getSelection of ESPUtil to change store
- override DojoGrid getSelected method
- conditionally add getSelected to widget params
- formatQuery return promise
- change loop in filterExists
- add a css grid-based version of form field grouping
- adding spray dialogs
- change GridForm to merge multiple groups of fields for state
- reorganizing some code for spray modals in LandingZone
- fix npm lint issues
- rearrange delimited form fields to make form a bit shorter
- revert changes to Filter component
- add react-hook-form as dependency
- some additional strings for LandingZone.tsx
- add FixedImportForm
- add Delimited, Json & Xml import forms
- add Variable & Blob import forms
- vscode sorted imports?
- additional constant for a generic required validation error
- removed unused import
- change TargetDropzone field to include the actual path as a property of
  each IDropdownOption object
- change TargetServer field to include OS as a property of the returned IDropdownOption objects
- correct types of TargetDfuSprayQueue field
- add TargetFolder field for LZ upload form
- fix lint error for async callback in useConst
- add dropzone for dragging files into LZ pane
- removing some debugging console logs
- add FileListForm to LandingZone
- removing some debug console logs & unused imports
- change how grid store and query are created / updated

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jeclrsg is this still draft or Ready for Review?

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+        isBlocking={false}
+        containerClassName={componentStyles.container}
+        dragOptions={dragOptions}
+    >
+        <div className={componentStyles.header}>
+            <span id={titleId}>{`${nlsHPCC.Import} ${nlsHPCC.Blob}`}</span>
+            <IconButton
+                styles={iconButtonStyles}
+                iconProps={cancelIcon}
+                ariaLabel="Close popup modal"
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:184]: Translations

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                    render={({
+                        field: { onChange, name: fieldName, value },
+                        fieldState: { error }
+                    }) => <TargetGroupTextField
+                        key="destGroup"
+                        label={nlsHPCC.Group}
+                        required={true}
+                        optional={true}
+                        selectedKey={value}
+                        placeholder={"Select a value"}
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:201]: Translations

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+                                    /> }
+                                    rules={{
+                                        required: nlsHPCC.ValidationErrorTargetNameRequired,
+                                        pattern: {
+                                            value: /^([a-z0-9]+[-a-z0-9 \._]+)+$/i,
+                                            message: nlsHPCC.ValidationErrorTargetNameInvalid
+                                        }
+                                    }}
+                                />
+                                <input type="hidden" { ...register(`selectedFiles.${idx}.SourceIP` as const) } value={file["NetAddress"]} />
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:283]: Expensive, foces re-refresh every time...  Looks like it shoould be a "useMemo"?

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+                            control={control} name="expireDays"
+                            render={({
+                                field: { onChange, name: fieldName, value },
+                                fieldState: { error }
+                            }) => <TextField
+                                name={fieldName}
+                                onChange={onChange}
+                                label={nlsHPCC.ExpireDays}
+                                value={value}
+                                errorMessage={ error && error.message }
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:361]: Change to error?.message

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+                console.log(err);
+            }
+        )();
+    }, [handleSubmit]);
+
+    const titleId = useId("title");
+
+    const dragOptions: IDragOptions = {
+        moveMenuItemText: "Move",
+        closeMenuItemText: "Close",
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:117]: Translations

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+                        key="destGroup"
+                        label={nlsHPCC.Group}
+                        required={true}
+                        optional={true}
+                        selectedKey={value}
+                        placeholder={"Select a value"}
+                        onChange={(evt, option) => {
+                            onChange(option.key);
+                        }}
+                        errorMessage={ error && error.message }
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:225]: Change to error?.message

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+                        key="DFUServerQueue"
+                        label={nlsHPCC.Queue}
+                        required={true}
+                        optional={true}
+                        selectedKey={value}
+                        placeholder={"Select a value"}
+                        onChange={(evt, option) => {
+                            onChange(option.key);
+                        }}
+                        errorMessage={ error && error.message }
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:246]: error?.message

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+                    /> }
+                    rules={{
+                        required: `Select a ${nlsHPCC.Queue}`
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:249]: Translations

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+                    render={({
+                        field: { onChange, name: fieldName, value },
+                        fieldState: { error }
+                    }) => <TextField
+                        name={fieldName}
+                        onChange={onChange}
+                        label={nlsHPCC.TargetScope}
+                        value={value}
+                        placeholder={nlsHPCC.NamePrefixPlaceholder}
+                        errorMessage={ error && error.message }
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:263]: ditto

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 10]
```diff
@@ -0,0 +1,497 @@
...
+                                    rules={{
+                                        required: nlsHPCC.ValidationErrorTargetNameRequired,
+                                        pattern: {
+                                            value: /^([a-z0-9]+[-a-z0-9 \._]+)+$/i,
+                                            message: nlsHPCC.ValidationErrorTargetNameInvalid
+                                        }
+                                    }}
+                                />
+                                <input type="hidden" { ...register(`selectedFiles.${idx}.SourceFile` as const) } value={file["fullPath"]} />
+                                <input type="hidden" { ...register(`selectedFiles.${idx}.SourceIP` as const) } value={file["NetAddress"]} />
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:304]: Expensive

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 11]
```diff
@@ -0,0 +1,497 @@
...
+                                    { key: "6", text: "UTF-16BE" },
+                                    { key: "7", text: "UTF-32" },
+                                    { key: "8", text: "UTF-32LE" },
+                                    { key: "9", text: "UTF-32BE" }
+                                ]}
+                                defaultSelectedKey="1"
+                                onChange={(evt, option) => {
+                                    onChange(option.key);
+                                }}
+                                errorMessage={ error && error.message }
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:337]: ditto

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/LandingZone.tsx [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+        setShowFileUpload(true);
+    }, [setShowFileUpload, setUploadFiles]);
+
+    return <HolyGrail
+        header={<CommandBar items={buttons} overflowButtonProps={{}} />}
+        main={
+            <>
+                <input
+                    id="uploaderBtn" type="file" accept="*.txt, *.csv, *.json, *.xml"
+                    style={{ display: "none" }} onChange={handleFileSelect} multiple={true}
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src-react/components/LandingZone.tsx:338]: @jeclrsg bump

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+            </Stack>
+            <Stack>
+                <table className={`${componentStyles.twoColumnTable} ${componentStyles.selectionTable}`}>
+                    <thead>
+                        <tr>
+                            <th>{nlsHPCC.SourcePath}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                    { selection.length && selection.map((file, idx) => {
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:216]: No need to test for length?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — DISMISSED** [review]

**Comment by richardkchapman** [comment]
Needs rebasing for a clean merge

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**